wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -486,17 +516,32 @@ private RecipeParser getRecipeParser(StageContext context)
    * Config for the plugin.
    */
   public static class Config extends PluginConfig {
-    static final String NAME_PRECONDITION = "precondition";
+    static final String NAME_PRECONDITION_JEXL = "preconditionJexl";
+    static final String NAME_PRECONDITION_SQL = "preconditionSql";
+    static final String NAME_PRECONDITION_LANGUAGE = "preconditionLanguage";
     static final String NAME_FIELD = "field";
     static final String NAME_DIRECTIVES = "directives";
     static final String NAME_UDD = "udd";
     static final String NAME_SCHEMA = "schema";
     static final String NAME_ON_ERROR = "on-error";
 
-    @Name(NAME_PRECONDITION)
+    @Name(NAME_PRECONDITION_LANGUAGE)
+    @Description("Toggle to configure precondition language between JEXL and SQL")
+    private String preconditionLanguage;
+
+    @Name(NAME_PRECONDITION_JEXL)
+    @Description("Precondition expression specifying filtering before applying directives (true to filter)")
+    @Macro
+    @Nullable
+    private String preconditionJexl;
```

Comment:
This is a backwards incompatible change. We should never change the name of existing properties.

This property must continue to be called `precondition`.